### PR TITLE
upgrade pyyaml version to make it compatible with python 3.12.2

### PR DESCRIPTION
**Pull request by Faraz32123**
pyyaml==6.0 is not compatible with python 3.12.
pyyaml==6.0.1 is compatible with python 3.8.0 to 3.12.2.
celery=4.x is also not compatible with 3.12.2. It throws below error
```
2024-04-15 12:36:32 Traceback (most recent call last):
2024-04-15 12:36:32   File "/openedx/venv/bin/celery", line 5, in <module>
2024-04-15 12:36:32     from celery.__main__ import main
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/celery/__init__.py", line 19, in <module>
2024-04-15 12:36:32     from . import local  # noqa
2024-04-15 12:36:32     ^^^^^^^^^^^^^^^^^^^
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/celery/local.py", line 17, in <module>
2024-04-15 12:36:32     from .five import PY3, bytes_if_py2, items, string, string_t
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/celery/five.py", line 7, in <module>
2024-04-15 12:36:32     import vine.five
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/vine/__init__.py", line 8, in <module>
2024-04-15 12:36:32     from .abstract import Thenable
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/vine/abstract.py", line 6, in <module>
2024-04-15 12:36:32     from .five import with_metaclass, Callable
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/vine/five.py", line 364, in <module>
2024-04-15 12:36:32     from inspect import formatargspec, getargspec as _getargspec  # noqa
2024-04-15 12:36:32     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-15 12:36:32 ImportError: cannot import name 'formatargspec' from 'inspect' (/opt/pyenv/versions/3.12.2/lib/python3.12/inspect.py). Did you mean: 'formatargvalues'?
```

backports-zoneinfo comes by-default in newer versions of python
it gives error while building wheel with python>=3.9
```
100.5 Downloading zope.interface-6.3-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (255 kB)
101.6    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 255.4/255.4 kB 223.8 kB/s eta 0:00:00
102.2 Building wheels for collected packages: backports-zoneinfo
102.2   Building wheel for backports-zoneinfo (pyproject.toml): started
102.8   Building wheel for backports-zoneinfo (pyproject.toml): finished with status 'error'
102.8   error: subprocess-exited-with-error
102.8
102.8   × Building wheel for backports-zoneinfo (pyproject.toml) did not run successfully.
102.8   │ exit code: 1
102.8   ╰─> [41 lines of output]
102.8       running bdist_wheel
102.8       running build
102.8       running build_py
102.8       creating build
102.8       creating build/lib.linux-x86_64-cpython-312
102.8       creating build/lib.linux-x86_64-cpython-312/backports
102.8       copying src/backports/__init__.py -> build/lib.linux-x86_64-cpython-312/backports
102.8       creating build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       copying src/backports/zoneinfo/_common.py -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       copying src/backports/zoneinfo/_version.py -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       copying src/backports/zoneinfo/_zoneinfo.py -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       copying src/backports/zoneinfo/__init__.py -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       copying src/backports/zoneinfo/_tzpath.py -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       running egg_info
102.8       writing src/backports.zoneinfo.egg-info/PKG-INFO
102.8       writing dependency_links to src/backports.zoneinfo.egg-info/dependency_links.txt
102.8       writing requirements to src/backports.zoneinfo.egg-info/requires.txt
102.8       writing top-level names to src/backports.zoneinfo.egg-info/top_level.txt
102.8       reading manifest file 'src/backports.zoneinfo.egg-info/SOURCES.txt'
102.8       reading manifest template 'MANIFEST.in'
102.8       warning: no files found matching '*.png' under directory 'docs'
102.8       warning: no files found matching '*.svg' under directory 'docs'
102.8       no previously-included directories found matching 'docs/_build'
102.8       no previously-included directories found matching 'docs/_output'
102.8       adding license file 'LICENSE'
102.8       adding license file 'licenses/LICENSE_APACHE'
102.8       writing manifest file 'src/backports.zoneinfo.egg-info/SOURCES.txt'
102.8       copying src/backports/zoneinfo/__init__.pyi -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       copying src/backports/zoneinfo/py.typed -> build/lib.linux-x86_64-cpython-312/backports/zoneinfo
102.8       running build_ext
102.8       building 'backports.zoneinfo._czoneinfo' extension
102.8       creating build/temp.linux-x86_64-cpython-312
102.8       creating build/temp.linux-x86_64-cpython-312/lib
102.8       gcc -pthread -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/openedx/venv/include -I/opt/pyenv/versions/3.12.2/include/python3.12 -c lib/zoneinfo_module.c -o build/temp.linux-x86_64-cpython-312/lib/zoneinfo_module.o -std=c99
102.8       lib/zoneinfo_module.c: In function ‘zoneinfo_fromutc’:
102.8       lib/zoneinfo_module.c:600:19: error: ‘_PyLong_One’ undeclared (first use in this function); did you mean ‘_PyLong_New’?
102.8         600 |             one = _PyLong_One;
102.8             |                   ^~~~~~~~~~~
102.8             |                   _PyLong_New
102.8       lib/zoneinfo_module.c:600:19: note: each undeclared identifier is reported only once for each function it appears in
102.8       error: command '/usr/bin/gcc' failed with exit code 1
102.8       [end of output]
102.8
102.8   note: This error originates from a subprocess, and is likely not a problem with pip.
102.8   ERROR: Failed building wheel for backports-zoneinfo
102.8 Failed to build backports-zoneinfo
102.8 ERROR: Could not build wheels for backports-zoneinfo, which is required to install pyproject.toml-based projects
------
Dockerfile:83
--------------------
  81 |
  82 |     # python requirements
  83 | >>> RUN --mount=type=cache,target=/openedx/.cache/pip,sharing=shared,uid=${APP_USER_ID} pip install -r requirements.txt
  84 |     # https://pypi.org/project/uWSGI/
  85 |     RUN --mount=type=cache,target=/openedx/.cache/pip,sharing=shared,uid=${APP_USER_ID} pip install uwsgi==2.0.24
--------------------
ERROR: failed to solve: process "/bin/sh -c pip install -r requirements.txt" did not complete successfully: exit code: 1
```

Upgraded Celery in this PR through constraints.txt which also resolves other dependencies conflicts with python 3.12.2.
This change is necessary to be merged to close this [issue](https://github.com/overhangio/tutor-ecommerce/issues/68) in tutor-ecommerce.

### CODE REVIEW

File: requirements/production.txt [hunk 1]
```diff
@@ -100,7 +100,7 @@ pytz==2023.3
     #   -r requirements/base.txt
     #   celery
     #   django
-pyyaml==6.0
+pyyaml==6.0.1
```
Comment (DawoudSheraz) [inline requirements/production.txt:131]: did you use `make upgrade` to update this? If not, it might get overridden in the next requirements upgrade run.

Comment (DawoudSheraz) [reply requirements/production.txt:131]: It seems https://github.com/openedx/ecommerce-worker/pull/246 is doing this

Comment (Faraz32123) [reply requirements/production.txt:131]: nope, i didn't use make upgrade.
#246 looks good, we can close this PR then.
Thanks for the update.

**Comment by Faraz32123** [comment]
closing this PR in favour of #246

File: requirements/production.txt (hunk 1 above)
Comment (Faraz32123) [reply requirements/production.txt:131]: @DawoudSheraz, #246 is closed without merge. Should I re-open this PR now.

Comment (DawoudSheraz) [reply requirements/production.txt:131]: This was closed automatically, a new PR has been created https://github.com/openedx/ecommerce-worker/pull/253 by requirements upgrade job.

Comment (Faraz32123) [reply requirements/production.txt:131]: Actually #253 doesn't include celery upgrade, which is necessary, all other modules of openedx have celery upgraded to 5.3.6. The celery version used in ecommerce-worker is celery==4.4.7 which is giving below error with python3.12.2:
```
2024-04-15 12:36:32 Traceback (most recent call last):
2024-04-15 12:36:32   File "/openedx/venv/bin/celery", line 5, in <module>
2024-04-15 12:36:32     from celery.__main__ import main
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/celery/__init__.py", line 19, in <module>
2024-04-15 12:36:32     from . import local  # noqa
2024-04-15 12:36:32     ^^^^^^^^^^^^^^^^^^^
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/celery/local.py", line 17, in <module>
2024-04-15 12:36:32     from .five import PY3, bytes_if_py2, items, string, string_t
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/celery/five.py", line 7, in <module>
2024-04-15 12:36:32     import vine.five
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/vine/__init__.py", line 8, in <module>
2024-04-15 12:36:32     from .abstract import Thenable
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/vine/abstract.py", line 6, in <module>
2024-04-15 12:36:32     from .five import with_metaclass, Callable
2024-04-15 12:36:32   File "/openedx/venv/lib/python3.12/site-packages/vine/five.py", line 364, in <module>
2024-04-15 12:36:32     from inspect import formatargspec, getargspec as _getargspec  # noqa
2024-04-15 12:36:32     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-15 12:36:32 ImportError: cannot import name 'formatargspec' from 'inspect' (/opt/pyenv/versions/3.12.2/lib/python3.12/inspect.py). Did you mean: 'formatargvalues'?
```

I have now upgraded the requirements using make upgrade which upgrades celery and its related dependencies. 
I have opened this PR or we can close it if they also upgrade celery.

File: requirements/base.txt [hunk 2]
```diff
@@ -1,82 +1,101 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.12
```
Comment (DawoudSheraz) [inline requirements/base.txt]: - we should use 3.8 environment to ensure the deps work with both 3.8 and 3.12

Comment (Faraz32123) [reply requirements/base.txt]: done.

File: requirements/constraints.txt [hunk 3]
```diff
@@ -18,4 +18,4 @@ Django<3.3
 -c common_constraints.txt
 
 # breaking changes in CLI and dropped support for python3.5 in celery 5.0
-celery<5.0
+celery<6.0.0
```
Comment (DawoudSheraz) [inline requirements/constraints.txt]: we should update the comment above the constraint.

**Review by DawoudSheraz — APPROVED** [review]

**Comment by DawoudSheraz** [comment]
@feanil Hi, need your assistance in getting this PR reviewed. Thanks

Also, I see codecov checks are stuck for other PRs as well. We can ignore it for now.

File: requirements/constraints.txt (hunk 3 above)
Comment (Faraz32123) [reply requirements/constraints.txt]: I think we do not need to place celery in constraints now, as it was placed here due to python3.5 in the first place.
Instead upgrading it to latest version will work.

**Comment by andrey-canon** [comment]
Hi @DawoudSheraz, @feanil I will make a review of this next week :D

**Comment by johanseto** [comment]
Hi everyone I was testing this PR, and I found some problems related to the tutor image creation. In this PR you are increasing Celery and that makes incompatible this line in the tutor ecommerce-worker image. 
https://github.com/overhangio/tutor-ecommerce/blob/f145bab7f3923cbcd765100c7e80becef5dbae2a/tutorecommerce/templates/ecommerce/build/ecommerce-worker/Dockerfile#L29




After I changed the command I could use the celery v5.4.0 in the worker.




image


Finally, I think this PR solves the problem. 
https://github.com/overhangio/tutor-ecommerce/pull/72/files#diff-774fa1744f2df2072567ed4736a42e704fd687c9b15f001d93c52b97756a5f0aR42

**Comment by johanseto** [comment]
Hi everyone I was testing this PR, and I found some problems related to the tutor image creation. In this PR you are increasing Celery and that makes incompatible this line in the tutor ecommerce-worker image. 
https://github.com/overhangio/tutor-ecommerce/blob/master/tutorecommerce/templates/ecommerce/build/ecommerce-worker/Dockerfile#L29




After I changed the command I could use the celery v5.

**Comment by johanseto** [comment]
Greetings to everyone, I took the courage to open this PR  in case it helps https://github.com/overhangio/tutor-ecommerce/pull/80

**Comment by DawoudSheraz** [comment]
@johanseto Hi. Yes, the PR to upgrade tutor-ecommerce plugin has been up for a while. It would be merged once this and openedx/ecommerce related PRs are merged.

**Review by feanil — APPROVED** [review]

**Comment by feanil** [comment]
I updated the repo to not require the codecov checks and removed the rev squad as a required reviewer.

**Pull request merged**